File: app/src/main/java/com/satanasov/phonebook/presenter/BasePresenter.java
package com.satanasov.phonebook.presenter;

public abstract class BasePresenter<T> {

    protected T view                   = null;
    protected boolean isSubscribed     = false;
    protected boolean isFirstSubscribe = true;

    public void attachView(T view){

        this.view = view;
    }

    public void subscribe(){

        isSubscribed = true;
    }

    public void unSubscribe(){
        isSubscribed     = false;
        isFirstSubscribe = false;
    }

    public void detachView(){

        this.view = null;
    }

}

File: app/src/main/java/com/satanasov/phonebook/kotlinPhoneBook/presenter/BasePresenter.kt
package com.satanasov.phonebook.kotlinPhoneBook.presenter

 abstract class BasePresenter<T> {
    protected var view: T?                  = null
    protected var isSubscribed: Boolean     = false
    protected var isFirstSubscribe: Boolean = true

    fun attachView(view: T){
        this.view = view
    }

    open fun subscribe(){
        isSubscribed = true
    }

    fun unSubscribe(){
        isSubscribed     = false
        isFirstSubscribe = false
    }

    fun detachView(){
        this.view = null
    }
}